test/mocks/v1/testing_snapshot_emitter.sk.go

```diff
@@ -227,204 +248,740 @@ func (c *testingEmitter) Snapshots(watchNamespaces []string, opts clients.WatchO
 		namespace string
 	}
 	mockCustomTypeChan := make(chan mockCustomTypeListWithNamespace)
-
 	var initialMockCustomTypeList MockCustomTypeList
 	/* Create channel for Pod */
 	type podListWithNamespace struct {
 		list      github_com_solo_io_solo_kit_pkg_api_v1_resources_common_kubernetes.PodList
 		namespace string
 	}
 	podChan := make(chan podListWithNamespace)
-
 	var initialPodList github_com_solo_io_solo_kit_pkg_api_v1_resources_common_kubernetes.PodList
 
 	currentSnapshot := TestingSnapshot{}
-	simplemocksByNamespace := make(map[string]SimpleMockResourceList)
-	mocksByNamespace := make(map[string]MockResourceList)
-	fakesByNamespace := make(map[string]FakeResourceList)
-	anothermockresourcesByNamespace := make(map[string]AnotherMockResourceList)
-	mctsByNamespace := make(map[string]MockCustomTypeList)
-	podsByNamespace := make(map[string]github_com_solo_io_solo_kit_pkg_api_v1_resources_common_kubernetes.PodList)
-
-	for _, namespace := range watchNamespaces {
-		/* Setup namespaced watch for SimpleMockResource */
-		{
-			simplemocks, err := c.simpleMockResource.List(namespace, clients.ListOpts{Ctx: opts.Ctx, Selector: opts.Selector})
+	simplemocksByNamespace := sync.Map{}
+	mocksByNamespace := sync.Map{}
+	fakesByNamespace := sync.Map{}
+	anothermockresourcesByNamespace := sync.Map{}
+	mctsByNamespace := sync.Map{}
+	podsByNamespace := sync.Map{}
+
+	if !watchNamespacesIsEmpty || opts.ExpressionSelector == "" {
```

Comment:
is this right? also rather than invert `watchNamespacesIsEmpty` we could just use `hasWatchedNamespaces`

Reply:
Why do you think this is incorrect? This is the backwards compatible.